Error on startup after clean install

Hey guys.

I just wanted to try Ki-nTree out and fired up a fresh xubuntu 20.04.1 VM. My problem is I can not start kintree.

kintree Version: 0.6.4
Python Version: 3.10.6
Installed via pip using virtual environment.

Error on startup:
`
(env-kintree) f@f-virtual-machine:~$ kintree
[ERROR] some Ki-nTree configuration files seem to be missing
`

I did not change the config files (nor did I copy a basic set). Am I supposed to do that before the first startup? I could not find it in the documentation :(

--- a/README.md
+++ b/README.md
@@ -123,7 +123,7 @@ Below is a summary table of the different configuration files, their function an
 | `parameters_filters.yaml` | Mapping between InvenTree parent categories and InvenTree parameters templates | :x: |
 | `search_api.yaml` | Generic controls for Supplier search APIs like cache validity and category matching ratio | :x: |
 | `supplier_parameters.yaml` | Mapping between InvenTree parameters templates and suppliers parameters/attributes, sorted by InvenTree parent categories (see [Part Parameters section](#part-parameters)) | :x: |
-| `<supplier>_config.yaml` | Mapping for supplier search results field names, to overwrite defaults (`<supplier>=['digikey', 'element14', 'lcsc', 'mouser']`) | :x: |
+| `<supplier>_config.yaml` | Mapping for supplier name and search results fields, to overwrite defaults (`<supplier>=['digikey', 'element14', 'lcsc', 'mouser']`) | :x: |
 | `<supplier>_api.yaml` | Required supplier API fields, custom to each supplier (`<supplier>=['digikey', 'element14', 'lcsc', 'mouser']`) | :heavy_check_mark: |
 | `digikey_categories.yaml` | Mapping between InvenTree categories and Digi-Key categories | :heavy_check_mark: |
 | `digikey_parameters.yaml` | Mapping between InvenTree parameters and Digi-Key parameters/attributes | :x: |
@@ -147,10 +147,11 @@ If you wish to automatically add subcategories while creating InvenTree parts, y
 Note that each time you enable the "Add" permission to an object, InvenTree automatically enables the "Change" permission to that object too.
 
 #### Settings
-1. With Ki-nTree GUI open, click on "Settings > Digi-Key" and fill in both Digi-Key API Client ID and Secret keys (optional: click on "Test" to [get an API token](#get-digi-key-api-token))
-2. Click on "Settings > Mouser" and fill in the Mouser part search API key
-3. Click on "Settings > KiCad", browse to the location where KiCad symbol, template and footprint libraries are stored on your computer then click "Save"
-4. If you intend to use InvenTree with this tool, click on "Settings > InvenTree" and fill in your InvenTree server address and credentials then click "Save" (optional: click on "Test" to get an API token)
+1. With Ki-nTree GUI open, click on "Settings > Supplier > Digi-Key" and fill in both Digi-Key API Client ID and Secret keys (optional: click on "Test" to [get an API token](#get-digi-key-api-token))
+2. Click on "Settings > Supplier > Mouser" and fill in the Mouser part search API key
+3. Click on "Settings > Supplier > Element14" and fill in the Element14 product search API key (key is shared with Farnell and Newark)
+4. Click on "Settings > KiCad", browse to the location where KiCad symbol, template and footprint libraries are stored on your computer then click "Save"
+5. If you intend to use InvenTree with this tool, click on "Settings > InvenTree" and fill in your InvenTree server address and credentials then click "Save" (optional: click on "Test" to get an API token)
 
 #### Get Digi-Key API token
 <details>
@@ -188,7 +189,7 @@ Refer to [this file](https://github.com/sparkmicro/Ki-nTree/blob/main/kintree/co
 
 #### Part Number Search
 
-Ki-nTree currently supports APIs for the following electronics suppliers: Digi-Key, Mouser and LCSC.
+Ki-nTree currently supports APIs for the following electronics suppliers: Digi-Key, Mouser, Element14 and LCSC.
 
 1. In the main window, enter the part number and select the supplier in drop-down list, then click "CREATE". It will start by fetching part data using the supplier's API
 2. In the case Digi-Key has been selected and the API token is not found or expired, a browser window will pop-up. To get a new token: [follow those steps](#get-digi-key-api-token)

--- a/kintree/config/settings.py
+++ b/kintree/config/settings.py
@@ -61,8 +61,8 @@ def load_user_config():
     CONFIG_USER_FILES = os.path.join(USER_SETTINGS['USER_FILES'], '')
 
     # Create user files folder if it does not exists
-    # if not os.path.exists(CONFIG_USER_FILES):
-    #     os.makedirs(CONFIG_USER_FILES)
+    if not os.path.exists(CONFIG_USER_FILES):
+        os.makedirs(CONFIG_USER_FILES)
     # Create user files
     return config_interface.load_user_config_files(path_to_root=CONFIG_ROOT,
                                                    path_to_user_files=CONFIG_USER_FILES,

--- a/kintree/database/inventree_api.py
+++ b/kintree/database/inventree_api.py
@@ -183,6 +183,14 @@ def is_new_part(category_id: int, part_info: dict) -> int:
     # cprint(filters)
 
     for part in part_list:
+        # Compare fields (InvenTree does not allow those to be identicals between two parts)
+        compare_fields = part_info['name'] == part.name and part_info['revision'] == part.revision
+        if compare_fields:
+            cprint(f'[TREE]\tWarning: Found part with same name and revision (pk = {part.pk})', silent=settings.SILENT)
+            return part.pk
+
+        # Compare parameters
+        compare_parameters = False
         # Get part parameters
         db_part_parameters = part.getParameters()
         part_parameters = {}
@@ -193,17 +201,12 @@ def is_new_part(category_id: int, part_info: dict) -> int:
 
         if new_part_parameters and part_parameters:
             # Compare database part with new part
-            compare = part_tools.compare(new_part_parameters=new_part_parameters,
-                                         db_part_parameters=part_parameters,
-                                         include_filters=filters)
-        else:
-            # Compare with name and description
-            compare = part_info['name'] == part.name and \
-                part_info['description'] == part.description and \
-                part_info['revision'] == part.revision
-            
-        if compare:
-            cprint(f'[TREE]\tWarning: Found part match in database (pk = {part.pk})', silent=settings.SILENT)
+            compare_parameters = part_tools.compare(new_part_parameters=new_part_parameters,
+                                                    db_part_parameters=part_parameters,
+                                                    include_filters=filters)
+                                                            
+        if compare_parameters:
+            cprint(f'[TREE]\tWarning: Found part with same parameters in database (pk = {part.pk})', silent=settings.SILENT)
             return part.pk
 
     # Check if manufacturer part exists in database
@@ -212,7 +215,7 @@ def is_new_part(category_id: int, part_info: dict) -> int:
     part_pk = is_new_manufacturer_part(manufacturer, mpn, create=False)
 
     if part_pk:
-        cprint(f'[TREE]\tWarning: Found part with same manufacturer part in database (pk = {part_pk})', silent=settings.SILENT)
+        cprint(f'[TREE]\tWarning: Found part with same manufacturer and MPN in database (pk = {part_pk})', silent=settings.SILENT)
         return part_pk
 
     cprint('\n[TREE]\tNo match found in database', silent=settings.HIDE_DEBUG)

--- a/kintree/database/inventree_interface.py
+++ b/kintree/database/inventree_interface.py
@@ -363,6 +363,8 @@ def inventree_fuzzy_company_match(name: str) -> str:
     inventree_companies = inventree_api.get_all_companies()
 
     for company_name in inventree_companies.keys():
+        cprint(f'{name.lower()} == {company_name.lower()} % {fuzz.partial_ratio(name.lower(), company_name.lower())}',
+               silent=settings.HIDE_DEBUG)
         if fuzz.partial_ratio(name.lower(), company_name.lower()) == 100:
             return company_name
     
@@ -448,10 +450,13 @@ def inventree_create(part_info: dict, categories: list, kicad=False, symbol=None
         if part_pk > 0:
             cprint('[INFO]\tPart already exists, skipping.', silent=settings.SILENT)
             ipn = inventree_api.get_part_number(part_pk)
-            # Update InvenTree part number
-            inventree_part['IPN'] = ipn
-            # Update InvenTree URL
-            inventree_part['inventree_url'] = f'{settings.PART_URL_ROOT}{inventree_part["IPN"]}/'
+            if ipn:
+                # Update InvenTree part number
+                inventree_part['IPN'] = ipn
+                # Update InvenTree URL
+                inventree_part['inventree_url'] = f'{settings.PART_URL_ROOT}{inventree_part["IPN"]}/'
+            else:
+                inventree_part['inventree_url'] = f'{settings.PART_URL_ROOT}{part_pk}/'
         # Part is new
         else:
             new_part = True
@@ -558,6 +563,8 @@ def inventree_create(part_info: dict, categories: list, kicad=False, symbol=None
         # Create manufacturer part(s)
         # TODO: Support multiple MPNs?
         for manufacturer_name, manufacturer_mpns in inventree_part['manufacturer'].items():
+            # Overwrite manufacturer name with matching one from database
+            manufacturer_name = inventree_fuzzy_company_match(manufacturer_name)
             # Get MPN
             manufacturer_mpn = manufacturer_mpns[0]
 
@@ -631,7 +638,7 @@ def inventree_create_alternate(part_info: dict, part_id='', part_ipn='', show_pr
         cprint('[INFO] Error: Original part was not found in database', silent=settings.SILENT)
         return result
 
-    # Overwrite manufacturer with matching one from database
+    # Overwrite manufacturer name with matching one from database
     manufacturer_name = inventree_fuzzy_company_match(part_info.get('manufacturer_name', ''))
     manufacturer_mpn = part_info.get('manufacturer_part_number', '')
     datasheet = part_info.get('datasheet', '')

--- a/kintree/kintree_gui.py
+++ b/kintree/kintree_gui.py
@@ -223,7 +223,7 @@ def element14_api_settings_window(supplier=''):
         [
             sg.Text(f'{supplier} Store', size=gui_global['big_label_size']),
             sg.Combo(sorted(element14_api.STORES[supplier]), default_value=default_store, key=f'{supplier.lower()}_store', enable_events=True),
-            sg.Text(f'URL: {element14_api.STORES[supplier][default_store]}', key='store_url'),
+            sg.Text(f'URL: {element14_api.STORES[supplier].get(default_store, "")}', key='store_url'),
         ],
         [
             sg.Button('Test', size=(15, 1)),
@@ -1246,6 +1246,9 @@ def main():
                             elif values['supplier'] == 'Mouser':
                                 error_message += '\n- Mouser API settings are correct ("Settings > Supplier > Mouser")' \
                                                  '\n- Part number is a valid number on Mouser\'s website'
+                            elif values['supplier'] in ['Farnell', 'Newark', 'Element14']:
+                                error_message += f'\n- {values["supplier"]} API key is correct and a store is selected ("Settings > Supplier > {values["supplier"]}")' \
+                                                 f'\n- Part number is a valid number on {values["supplier"]}\'s website'
                             elif values['supplier'] == 'LCSC':
                                 error_message += '\n- Mouser API settings are correct ("Settings > Supplier > LCSC")' \
                                                  '\n- Part number is a valid LCSC number (eg. starts with "C")'

